client/src/app/pages/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { BehaviorSubject, Observable, Subject, Subscription } from 'rxjs';
+import { AddTask, EditTask, Task } from './models/models.tasks';
+import { TasksService } from './services/tasks.service';
+import { cloneDeep } from 'lodash';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss'],
+})
+export class DashboardComponent implements OnInit, OnDestroy {
+  tasksList = new BehaviorSubject<Task[] | null>(null);
+
+  tasksCompleted: Task[] = [];
+  tasksInProgress: Task[] = [];
+
+  subscriptionTasksMany: Subscription;
+
+  constructor(public tasksService: TasksService) {
+    this.subscriptionTasksMany = this.tasksList.subscribe((e) => {
+      if (e === null) return;
+
+      this.tasksCompleted = e.filter(
+        (element) => element.status.value === 'completed'
+      );
+
+      this.tasksCompleted.sort((a, b) => b.status.index - a.status.index);
+
+      this.tasksInProgress = e.filter(
+        (element) => element.status.value === 'in-progress'
+      );
+
+      this.tasksInProgress.sort((a, b) => b.status.index - a.status.index);
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptionTasksMany.unsubscribe();
+  }
+
+  ngOnInit(): void {
+    const a = this.tasksService.getTasksByStatus('in-progress');
+    const b = this.tasksService.getTasksByStatus('completed');
+
+    const cloned = cloneDeep(a.concat(b));
+
+    this.tasksList.next(cloned);
+  }
+
+  addTaskEvent(myData: AddTask) {
+    const newTask = this.tasksService.addTask(myData);
+
+    const oldValue = this.tasksList.value;
+
+    if (oldValue === null) return;
+
+    const myTasks = [newTask, ...oldValue];
+
+    this.tasksList.next(myTasks);
+  }
+
+  deleteTaskEvent(_id: string) {
```

Comment:
You have a lot of logic inside component. WE should use services for it

Reply:
Should i store the state in a service?